### PR TITLE
Hotfix/appeals 23420 v2

### PR DESCRIPTION
**Pull request by AimanK**
This branch should resolve and localize [APPEALS-23420](https://jira.devops.va.gov/browse/APPEALS-23420)

# Description
This search defect timeout is reached prior to returned results. There is no reportable issue/error coming from the logs. Only client side error as show in the screenshot below. This is due to the high amount of code execution time in VACOLS, and the unoptimized serializers being used that cause the 60 second timeout to expire and the client code throws an error stating that a server error has occurred. After an initial fix to the serializer globally, too many downstream errors in the system were experienced by users limited by the new serializer elsewhere wihin Caseflow. This new code should localize this defect code to the case search page.

## Before/Error message
image

## Expected results
image

## Acceptance Criteria
- [x] Code compiles correctly
- [x] Previously failing veteran case searches now beat the timeout by 30 seconds
- [ ] No downstream errors are found elsewhere within Caseflow (Hearings Org, IDT, and Case Details error)

## Testing Plan
- [x] Successful testing in Demo
- [x] Successful testing in UAT
- [ ] Successful testing in Prodtest

### CODE REVIEW

File: client/app/queue/CaseList/CaseListActions.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import * as Constants from './actionTypes';
 
 import { get, size } from 'lodash';
 import { onReceiveAppealDetails, onReceiveClaimReviewDetails } from '../QueueActions';
-import { prepareAppealForStore, prepareClaimReviewForStore } from '../utils';
+import { prepareAppealForSearchStore, prepareClaimReviewForStore } from '../utils';
```
Comment (raymond-hughes) [inline client/app/queue/CaseList/CaseListActions.js:7]: Should we be using a separate Store here and leaving the original or is this going to be ok to do?

File: client/app/queue/CaseList/CaseListActions.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ export const fetchedNoAppeals = (searchQuery) => ({
 });
 
 export const onReceiveAppeals = (appeals) => (dispatch) => {
-  dispatch(onReceiveAppealDetails(prepareAppealForStore(appeals)));
+  dispatch(onReceiveAppealDetails(prepareAppealForSearchStore(appeals)));
```
Comment (raymond-hughes) [inline client/app/queue/CaseList/CaseListActions.js:57]: I want to make sure we don't share this with anything else

File: client/app/queue/VeteranCasesView.jsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ import { setFetchedAllCasesFor } from './CaseList/CaseListActions';
 import { hideVeteranCaseList } from './uiReducer/uiActions';
 import { onReceiveAppealDetails } from './QueueActions';
 import { appealsByCaseflowVeteranId } from './selectors';
-import { prepareAppealForStore } from './utils';
+import { prepareAppealForSearchStore } from './utils';
```
Comment (raymond-hughes) [inline client/app/queue/VeteranCasesView.jsx:15]: Same here, is this sharing the serializer with anything else?

File: client/app/queue/VeteranCasesView.jsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ class VeteranCasesView extends React.PureComponent {
           return Promise.reject(response);
         }
 
-        this.props.onReceiveAppealDetails(prepareAppealForStore(returnedObject.appeals));
+        this.props.onReceiveAppealDetails(prepareAppealForSearchStore(returnedObject.appeals));
```
Comment (raymond-hughes) [inline client/app/queue/VeteranCasesView.jsx:53]: Verify this one as well. Lets set up a chat tomorrow

File: client/app/queue/utils.js [hunk 5]
```diff
@@ -568,22 +572,19 @@ export const prepareAppealForSearchStore = (appeals) => {
       appellantMiddleName: appeal.attributes.appellant_middle_name,
       appellantLastName: appeal.attributes.appellant_last_name,
       appellantSuffix: appeal.attributes.appellant_suffix,
-      appellantDateOfBirth: appeal.attributes.appellant_date_of_birth,
-      appellantAddress: appeal.attributes.appellant_address,
-      appellantEmailAddress: appeal.attributes.appellant_email_address,
-      appellantPhoneNumber: appeal.attributes.appellant_phone_number,
```
Comment (raymond-hughes) [inline client/app/queue/utils.js:574]: Is anything else using this utils module? I don't want to have a situation where some other class/module is using these but we are removing them

**Review by raymond-hughes — CHANGES_REQUESTED** [review]

**Comment by raymond-hughes** [comment]
The tests look good so far. We had one failure but I think rerunning it will solve the issue

File: client/app/queue/CaseList/CaseListActions.js (hunk 2 above)
Comment (AimanK) [reply client/app/queue/CaseList/CaseListActions.js:57]: Confirmed that this action is only used on case search

File: client/app/queue/VeteranCasesView.jsx (hunk 3 above)
Comment (AimanK) [reply client/app/queue/VeteranCasesView.jsx:15]: Confirmed that this is isolated and correct

File: client/app/queue/VeteranCasesView.jsx (hunk 4 above)
Comment (AimanK) [reply client/app/queue/VeteranCasesView.jsx:53]: Verified that this is correct

File: client/app/queue/utils.js (hunk 5 above)
Comment (AimanK) [reply client/app/queue/utils.js:574]: Gotcha, the way that this prepareAppealsForSearchStore hash has been set up, it should only be used on the two case search pages (within case details, and the main case search page), and accordingly, it should only render the necessary attributes. I went one by one and verified specifically which attributes were needed vs. not needed as well.

**Pull request merged**